Source/dom/export.js

```diff
@@ -46,58 +46,146 @@ export const DEFAULT_EXPORT_OPTIONS = {
  *
  *
  * @param {dictionary} options Options indicating which sizes and formats to use, etc.
+ *
+ * @returns If an output path is not set, the data is returned
  */
 export function exportObject(object, options) {
-  const merged = { ...DEFAULT_EXPORT_OPTIONS, ...options }
+  if (!object) {
+    throw new Error('No object provided to export')
+  }
+  let exportJSON = false
+  let { formats } = options
+
+  if (typeof formats === 'string') {
+    formats = formats
+      .split(',')
+      .map(format => format.trim())
+      .filter(t => t.length > 0)
+  }
+  if (formats) {
+    const idx = formats.indexOf('json')
+    if (idx >= 0) {
+      formats.splice(idx, 1)
+      exportJSON = true
+    }
+    formats = formats.join(',')
+  }
+
+  const merged = {
+    ...DEFAULT_EXPORT_OPTIONS,
+    ...options,
+    formats,
+  }
+  const objectsToExport = (Array.isArray(object) ? object : [object]).map(
+    o => (isWrappedObject(o) ? o.sketchObject : o)
+  )
+  if (!objectsToExport) {
+    throw new Error('No objects provided to export')
+  }
+
+  function archiveNativeObject(nativeObject) {
+    const archiver = MSJSONDataArchiver.new()
+    archiver.archiveObjectIDs = true
+    const aPtr = MOPointer.alloc().init()
+    const obj = nativeObject.immutableModelObject
+      ? nativeObject.immutableModelObject()
+      : nativeObject
+    archiver.archivedDataWithRootObject_error(obj, aPtr)
+    if (aPtr.value()) {
+      throw Error('Archive error')
+    }
+    return archiver.archivedData()
+  }
+
+  // Return data if no output directory specified
+  if (!merged.output) {
+    const firstObject = objectsToExport[0]
```

Comment:
Why only the first object? Let’s export all of them and return an array

Reply:
That felt a little weird in the case of the image data but I agree it is the less confusing option. I'll fix it

Reply:
Looking to the future, any thoughts on multiple formats? My instinct is to limit it to a single type if `!output`. The only good response I can think of otherwise is an array of objects like:
```
[ 
    { 'jpg' : <data>, png: <data> },
    { 'jpg' : <data>, png: <data> }
]
```

Reply:
That’s a good question, I was actually thinking about returning an NSImage so it wouldn’t care about the format.

So let’s do this: if there are more than one format and output is falsy, then we throw an error. We only authorise 1 format if there is no output